platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -878,9 +878,25 @@ + (instancetype)expressionWithMGLJSONObject:(id)object {
                 NSExpression *expression = [NSExpression expressionWithMGLJSONObject:argumentObjects[index]];
                 NSMutableDictionary *attrs = [NSMutableDictionary dictionary];
                 if ((index + 1) < argumentObjects.count) {
-                    attrs = argumentObjects[index + 1];
+                    attrs = [NSMutableDictionary dictionaryWithDictionary:argumentObjects[index + 1]];
                 }
                 
+                if (attrs.count) {
+                    if (attrs[MGLFontNamesAttribute]) {
+                        NSArray *fontNames = (NSArray *)attrs[MGLFontNamesAttribute];
+                        attrs[MGLFontNamesAttribute] = fontNames[1];
+                    }
+                    if (attrs[MGLFontColorAttribute] && [attrs[MGLFontColorAttribute] isKindOfClass:[NSArray class]]) {
+                        NSArray *colorArray = attrs[MGLFontColorAttribute];
+                        if ([colorArray[0] isEqualToString:@"rgb"] || [colorArray[0] isEqualToString:@"rgba"]) {
```

Comment:
This will be easy to overlook when we add additional color-generating operators like `hsl` and `hsla`. Does GL JS do anything fancy to centralize color parsing?

Reply:
I'm not sure. Although in the docs I specify that you should pass an MGLColor value, users can provide an hsla value, or `yellow`, or `rgba(255, 255, 255, 1)` string, but to keep consistency I'm specifying they should use of `MGLColor`.

Reply:
Although we can refactor this once we add more colors into a color parsing function.

Reply:
So we’re documenting the attribute as taking only a UIColor/NSColor, but in reality we also accept JSON-style `rgb`/`rgba` arrays? In that case, I think we should either remove this bit of code or officially support the arrays. If we do support the arrays, we could just include the array verbatim in the overall expression, relying on mbgl to raise an error if the developer tries to be clever and stick an actual expression in there.

Reply:
Using the straight color syntax from the spec creates a consistency problem when comes to color. On iOS/macos colors are passes as a range of `0-1.0` but in our spec are `0-255` we still don't support the other formats oficially, it may make more sense to restrict the value to `MGLColor` then.

Reply:
Oh, I just meant as an alternative to UIColor/NSColor – the native type would be preferable for sure.